snapcraft/internal/project_loader/grammar/_to.py

```diff
@@ -146,4 +157,6 @@ def _extract_to_clause_selectors(to):
         raise ToStatementSyntaxError(
             to, message='spaces are not allowed in the selectors')
 
-    return {selector.strip() for selector in selector_group.split(',')}
+    return (
+        {selector.strip() for selector in on_selector_group[3:].split(',')},
```

Comment:
This `3` is yucky. I know it's for the `on\s` but I'll probably forget. Would you mind adding a comment for that, please? Also, returning a set containing an empty string for a statement that's completely missing does not seem to be a clear API to me. I would expect it to return e.g. None. Thoughts?